crates/mirror-mirror/src/value.rs

```diff
@@ -138,7 +140,16 @@ impl PartialOrd for Value {
 
 impl Ord for Value {
     fn cmp(&self, other: &Self) -> Ordering {
-        OrdEqValue::from(self).cmp(&OrdEqValue::from(other))
+        OrdEqHashValue::from(self).cmp(&OrdEqHashValue::from(other))
+    }
+}
+
+impl Hash for Value {
+    fn hash<H>(&self, state: &mut H)
+    where
+        H: Hasher,
+    {
+        core::mem::discriminant(&OrdEqHashValue::from(self)).hash(state);
```

Comment:
Why the `discriminant` call? Can we let the converted type do all the hashing itself?

```suggestion
                OrdEqHashValue::from(self).hash(state);
```

Reply:
Good idea! I just had rust-analyzer expand the derived `Hash` impl and got this. Actually don't fully understand what `discriminant` does. But your suggestion is indeed nicer!